src/ui/prompts/listPrompts/targetingListPrompt.ts

```diff
@@ -0,0 +1,97 @@
+import { ListOption, ListOptionsPrompt } from './listOptionsPrompt'
+import inquirer from 'inquirer'
+import { AddItemPrompt, EditItemPrompt, RemoveItemPrompt, ContinuePrompt, ExitPrompt } from './promptOptions'
+import { servePrompt, audienceNamePrompt } from '../targetingPrompts'
+import { Filters, UpdateTargetParams } from '../../../api/schemas'
+import { FilterListOptions } from './filterListPrompt'
+import Writer from '../../writer'
+
+export class TargetingListOptions extends ListOptionsPrompt<UpdateTargetParams> {
+    itemType = 'Targeting Rule'
+    messagePrompt = 'Manage your Targeting'
+    authToken: string
+    projectKey: string
+    featureKey: string
+
+    constructor(list: UpdateTargetParams[], writer: Writer, authToken: string, projectKey: string, featureKey: string) {
+        super(list, writer)
+        this.featureKey = featureKey
+        this.authToken = authToken
+        this.projectKey = projectKey
+    }
+
+    options() {
+        return [
+            AddItemPrompt(this.itemType),
+            EditItemPrompt(this.itemType),
+            RemoveItemPrompt(this.itemType),
+            ContinuePrompt,
+            ExitPrompt
+        ]
+    }
+
+    async promptAddItem<UpdateTargetParams>(): Promise<ListOption<UpdateTargetParams>> {
+        const { name, serve } = await inquirer.prompt([audienceNamePrompt, servePrompt], {
+            token: this.authToken,
+            projectKey: this.projectKey,
+            featureKey: this.featureKey
+        })
+        const filters: Filters = await (new FilterListOptions([], this.writer)).prompt()
+        const target = {
+            distribution: [{ _variation: serve.key, percentage: 1 }],
+            audience: { name, filters: { filters, operator: 'and' } }
+        }
+
+        return {
+            name: target.audience.name,
+            value: target as UpdateTargetParams
+        }
+    }
+
+    async promptEditItem<UpdateTargetParams>(
+        list: ListOption<UpdateTargetParams>[]
+    ): Promise<void> {
+        if (list.length === 0) {
+            this.writer.warningMessage(`No ${this.itemType}s to edit`)
+            return
+        }
+
+        const response = await inquirer.prompt([{
+            name: 'targetToEdit',
+            message: `Which ${this.itemType} would you like to edit?`,
+            type: 'list',
+            choices: list
+        }])
+        const index = list.findIndex((item) => item.name === response.targetToEdit.audience.name)
```

Comment:
matching based on the name seems unreliable, since there's nothing preventing targets from having duplicate names

Reply:
yeah, I agree. I don't really see a way to uniquely identify targets. I could stringify the entire target and match based on that value but even then there's nothing that stops someone from having the same defintion for a target

Reply:
all the targets fetched from the API should have a unique `_id`

Reply:
So, yeah I can use the `_id` property if it exists but what can I reliably use if doesn't? Newly added targets in the list won't have `_id`

Reply:
The same thing just came up in Emma's PR. My suggestion was to pass along the `index` as part of the response object so you dont need to use `findIndex` to update the item. Think that might work?

Reply:
Yeah I think that could work. But I'm gonna do it as apart of another ticket. The reason being that I'd need to change the definition for `transformToListOptions` which means I'd need to update all the other listPrompt classes. This PR is already big enough as it is